json/src/test/java/org/apache/dto/FirstDTO.java

```diff
@@ -33,17 +34,20 @@
     private final Optional<Long> id;
     private final String time;
     private final String payload;
+    private final Optional<DTO> child;
 
     @JsonCreator
     public FirstDTO(
             @JsonProperty("type") String type,
             @JsonProperty("id") Optional<Long> id,
             @JsonProperty("time") String time,
-            @JsonProperty("payload") String payload) {
+            @JsonProperty("payload") String payload,
+            @JsonProperty("child") Optional<DTO> child) {
```

Comment:
it's a nested DTO. Why would you want a list or a set?

Reply:
because a DTO could have several nested DTO. This is also related to @remk's question: why defining the DTO as nested/child DTO, and not defining all the DTO modules at the same level?

Reply:
I think there's a misunderstanding: FirstDTO is just a DTO for test. It aims at exercising the nested DTO serialization but proving that we can serialize List<DTO> instead of Optional<DTO> brings nothing to the test